processcheck/topProcess.py

```diff
@@ -1,10 +0,0 @@
-def topProcess(numOfProcesses):
-    try:
-        import psutil
-        print('psutil found')
-    except ImportError:
-        raise RuntimeError('psutil missing!')
-    from pprint import pprint as pp
-
-    #using pretty print to organize to data; also sorts processes based on size: biggest first, smallest last
-    pp([(p.info['name'] + "(" + str(p.pid) + "): {0:.2f}".format(p.info['memory_info'].rss / 1024 / 1024) + " MB") for p in sorted(psutil.process_iter(attrs=['name', 'memory_info']), key=lambda p: p.info['memory_info'])][-numOfProcesses:]) 
```

Comment:
The Ubuntu servers and my MacBook had difficulty with this line.  I would consistently get that process_iter requires no arguments and it had 1.  

If I switch to the older .as_dict method which is what the shorthand does internally it works fine.

Reply:
It may be nice for this definition to return a list of dictionary pairs for consumption into another script.   As is we would probably have to encode the output and convert it to a dictionary before we could perform any logic against it.